Fix cucumber tests and add eslint

stringhandler opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
- Fix cucumber tests
- Add ESLint to integration tests
- Add ESLint to CI run

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
To fix the tests.
Added ESLint to add standards for javascript

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [] I have squashed my commits into a single commit.

delta1 commented:
Nice definitely needed

delta1 commented on .circleci/config.yml:
```diff
@@ -92,6 +92,9 @@ commands:
          command: cd integration_tests && npm run check-fmt
+     - run:
+         name: Check eslint
+         command: cd integration_tests && npx eslint .
```
any reason to not add eslint to package.json with a script there?

stringhandler replied:
lol, because my js is not great... that is the reason

delta1 commented on integration_tests/cucumber.js:
```diff
@@ -1,4 +1,4 @@
-};
+  default: '',
+  critical: '--format @cucumber/pretty-formatter --tags @critical'
+}
```
I think there's a conflict between prettier and eslint here (single quotes vs double quotes, and semi-colons)

delta1 reviewed (commented):
I think we just need to resolve differences between prettier and eslint

Commit:
Fix cucumber test

Commit:
[tests] eslint config and husky pre-commit hook

- update eslint config to play nicely with prettier
- fix lint issues
- add husky to run pre-commit hook
- add pre-commit hook to run prettier and lint check if files in
  integration_tests have been staged for commit

Tested on macOS and windows

stringhandler committed:
Merge pull request #4 from delta1/eslint

[tests] eslint config and husky pre-commit hook
